xgcm/grid.py

```diff
@@ -1535,6 +1704,57 @@ def average(self, da, axis, **kwargs):
         # do we need to pass kwargs?
         return weighted.sum(dim, **kwargs) / weight.sum(dim, **kwargs)
 
+    def transform(self, da, axis, target, **kwargs):
+        """Convert an array of data to new 1D-coordinates.
+        The method takes a multidimensional array of data `da` and transforms it onto another data_array `target_data` in the direction of the axis (for each 1-dimensional 'column').
+
+        `target_data` can be e.g. the existing coordinate along an axis, like depth. xgcm automatically detects the appropriate coordinate and then
+        transforms the data from the input positions to the desired positions defined in `target`. This is the default behavior.
+        The method can also be used for more complex cases like transforming a dataarray into new coordinates that are defined by e.g. a tracer field
+        like temperature, density, etc.
+
+        Currently two methods are supported to carry out the transformation:
+            - 'linear': Values are linear interpolated between 1D columns along `axis` of `da` and `target_data`.
+                        This method requires `target_data` to increase/decrease monotonically. `target` values are interpreted as new cell centers in this case.
+                        By default this method will return nan for values in `target` that are outside of the range of `target_data`, setting `mask_edges=False` results in the default
+                        np.interp behavior of repeated values.
+
+            - 'conservative':   Values are transformed while conserving the integral of `da` along each 1D column. This method can be used with non-monotonic values of `target_data`.
+                                Currently this will only work with extensive quantities (like heat, mass, transport) but not with intensive quantities (like temperature, density, velocity).
+                                N given `target` values are interpreted as cell-bounds and the returned array will have N-1 elements along the newly created coordinate,
+                                with coordinate values that are interpolated between `target` values.
```

Comment:
This docstring may look okay online, but it renders very poorly on the command line in fixed-width mode. Can you reformat it to use shallower indentation? I don't think you have to align things so far right.

Reply:
Is there a linter for docstrings that would take care of this? was wondering about that for a while.

Reply:
https://pypi.org/project/flake8-docstrings/

Reply:
https://github.com/pycqa/pydocstyle

Reply:
awesome. Will raise a seperate issue to integrate those with pre-commit.

Reply:
Hmm black does not like the slimmer version...